nix/latex.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, lib, texlive }:
+{
+  # Build a latex derivation using latexmk.
+  buildLatex = { 
+    texFiles ? [], # The specific tex files to build, will try and build all of them if absent
+    texInputs ? { inherit (texlive) scheme-small; }, # Tex dependencies as an attrset
+    buildInputs ? [], # Additional build inputs
+    ...}@attrs :
+
+    let 
+      tex = texlive.combine (texInputs // { inherit (texlive) latexmk; });
+      # mkDerivation doesn't like having this as an attr, and we don't need to pass it through
+      filteredAttrs = builtins.removeAttrs attrs ["texInputs" ];
+      buildDir = ".nix-build";
+    in
+    stdenv.mkDerivation (filteredAttrs // {
+      buildInputs = [ tex ] ++ buildInputs;
+      buildPhase = ''
+        runHook preBuild
+        mkdir -p ${buildDir}
+        latexmk -outdir=${buildDir} -pdf ${toString texFiles}
+        runHook postBuild
+      '';
+      installPhase = ''
+        runHook preInstall
+        mkdir -p $out
+        install -t $out ${buildDir}/*.pdf
+
+        mkdir -p $out/nix-support
+        for pdf in $out/*.pdf; do
+          echo "doc-pdf pdf $pdf" >> $out/nix-support/hydra-build-products
```

Comment:
what's this stuff for?

Reply:
It makes it so the hydra jobs have a link to download the build output on them directly, which is neat for linking people to the PDFs.

Reply:
oh nice, I was looking for something like that ages ago.